x-pack/plugins/ml/server/routes/system.ts

```diff
@@ -277,6 +277,52 @@ export function systemRoutes(
             return acc;
           }, {} as Record<string, { exists: boolean }>);
 
+          return response.ok({
+            body: result,
+          });
+        } catch (error) {
+          return response.customError(wrapError(error));
+        }
+      })
+    );
+
+  /**
+   * @apiGroup SystemRoutes
+   *
+   * @api {post} /internal/ml/reindex_with_pipeline ES reindex wrapper to reindex with pipeline
+   * @apiName MlReindexWithPipeline
+   */
+  router.versioned
+    .post({
+      path: `${ML_INTERNAL_BASE_PATH}/reindex_with_pipeline`,
+      access: 'internal',
+      options: {
+        tags: ['access:ml:canGetMlInfo'],
+      },
+    })
+    .addVersion(
+      {
+        version: '1',
+        validate: {
+          request: {
+            body: schema.object({
```

Comment:
As this is just a basic wrapper for `reindex` I wonder if having any route validation is needed?
The real request payload to `reindex` is `ReindexRequest` and it's a lot more complicated than this schema. 
This schema could incorrectly reject a legitimate payload.

Reply:
Updated in 78af285e55c9eef7203b9d88ee617ea306cf460a